### ISSUE TITLE
Github Repo theme folder name converted to lowercase

### ISSUE THREAD

**Issue by bradmkjr** [afragen/github-updater#291]
I hope someone else can replicate this issue.

I used the plugin (develop branch) install to install a custom WP theme, from a private Github repo. The repository slug has Capital Letters. When I initially install a plugin it seems to convert the slug to lowercase. When I go to update the theme, it adds a 2nd folder with proper Capitalization.

I searched the repository for strtolower, and don't see anywhere that would be attempting to change the capitalization to lowercase, so it maybe an issue with WP core.

Any suggestions or ideas?

**Comment by afragen** [comment]
Can you give `read` access to `afragen` and share the URI so I can test? Not sure I've even run across a mixed case repo name.

`strtolower`does exist in `Base::get_repo_parts`

**Comment by bradmkjr** [comment]
Thank you,

URI is
https://github.com/DesignMissoula/DM-theme

Branch in question:
sunrivergourmet.com

https://github.com/DesignMissoula/DM-theme/tree/sunrivergourmet.com

**Comment by afragen** [comment]
You'll need to give me read access or email me an access token for the repo.

**Comment by bradmkjr** [comment]
Ok, I did invite you to the Design Missoula team, but here is a API key for
you:

2bb1490632076b739fe0b018201b2b0213e19516

Thanks for looking into this for me.

**Comment by afragen** [comment]
So far I think I've discovered that `sanitize_key()` uses `strtolower()` and so the $options for a mixed case repo get messed up. This means that the access token never gets saved properly.

I can revert this change in `Settings::sanitize()` and the token saves.

I have been able to install both FoundationPress (parent theme) and your specific theme and branch. I even decremented the local version number to update and the update proceeded correctly.

I've just pushed a commit/bump to `develop`. Update and let me know if the lowercase switch happens.
